error[E0152]: found duplicate lang item `owned_box`

 --> mutant.rs:3:1

  |

3 | struct Foo ;

  | ^^^^^^^^^^^^

  |

  = note: the lang item is first defined in crate `alloc` (which `std` depends on)

  = note: first definition in `alloc` loaded from /home/cnsun/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-3aeb407930ebd519.rlib

  = note: second definition in the local crate (`mutant`)



thread 'rustc' panicked at 'index out of bounds: the len is 0 but the index is 0', compiler/rustc_middle/src/ty/subst.rs:302:43

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace



error: internal compiler error: unexpected panic



note: the compiler unexpectedly panicked. this is a bug.



note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md



note: rustc 1.53.0-nightly (07e0e2ec2 2021-03-24) running on x86_64-unknown-linux-gnu



note: compiler flags: -Z mir-opt-level=3 -Z new-llvm-pass-manager=yes -Z unsound-mir-opts=yes -Z polonius=yes -Z polymorphize=yes -C debuginfo=2 -C opt-level=3 --crate-type staticlib



query stack during panic:

#0 [typeck] type-checking `main`

#1 [typeck_item_bodies] type-checking all item bodies

end of query stack

error: aborting due to previous error



For more information about this error, try `rustc --explain E0152`.

